src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/InstrumentAdvice.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace System.Diagnostics.Metrics
+{
+    /// <summary>
+    /// Contains configuration settings advised to be used by metrics consumers when recording measurements for a given <see cref="Instrument{T}"/>.
+    /// </summary>
+    /// <typeparam name="T">Instrument value type.</typeparam>
+    public sealed class InstrumentAdvice<T> where T : struct
```

Comment:
Why must T be a value type?

Reply:
Oh, I see, it's not that it needs to be a value type, it's that it must be just one of byte, short, int, long, double, float, or decimal. Do we use such a constraint in other places for the same purpose?

Reply:
We support only specific set of types 

https://github.com/dotnet/runtime/blob/ced2117d24e21c145c03749ea2b198ea08638323/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Instrument.cs#L141-L149.

The supported `T` types in the advice class should exactly match the same supported types in `Instrument<T>`

Reply:
> Do we use such a constraint in other places for the same purpose?

I am not aware of other places other than metrics. @tannergooding may know if there are other places that apply similar restrictions.

Reply:
But we already use the same constraint for the same limited set of types elsewhere in metrics?

Reply:
> But we already use the same constraint for the same limited set of types elsewhere in metrics?

Correct.